src/main/java/com/auth0/jwk/JwkSet.java

```diff
@@ -0,0 +1,78 @@
+package com.auth0.jwk;
+
+import java.util.Collections;
+import java.util.List;
+import static com.google.common.base.Preconditions.checkArgument;
+
+/**
+ * Represents a set of JSON Web Keys (JWK) used to verify the signature of JWTs.
+ */
+public class JwkSet {
+
+    private final List<Jwk> keys;
+
+    /**
+     * Create a new empty JWK set.
+     */
+    public JwkSet() {
+        this(Collections.<Jwk>emptyList());
+    }
+
+    /**
+     * Creates a new JWK set with a single key.
+     * @param key the JWK. Must not be {@code null}.
+     */
+    public JwkSet(final Jwk key) {
```

Comment:
Normally, a Set would be created from a JSON array after parsing the jwks file, for that scenario I'd only keep 1 method signature: the one that accepts the list/set. This one I'd remove.

What is the planned use case?

Reply:
The use case will be this:
* The `GuavaCachedJwkProvider` will be updated to cache the JWKS response, which will be this new `JwkSet` object. 
* A `getSet()` method will be added to the `JwkProvider` instance, which will allow the cached provider to get the entire JWKS response if the cache is empty or the requested key is not found.
* The `getSet()` implementation in `UrlJwkProvider` will create a new `JwkSet` from the existing `getAll()` method.

To your point, this constructor can be removed. It was added just as general practice and ease-of-use, but it is not critical and can be removed and added later if there is a specific requirement.

Reply:
Thanks for the details. You mentioned `getSet()` would be called when the cache is empty or key not found, is this the method where the actual request is made?


And yes, let's remove this one, as you said we can add it later if needed.

Reply:
`getSet()` will use the existing `getAll()` method, which is where the actual request is made.